google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java

```diff
@@ -176,6 +182,8 @@ private static SpannerException newSpannerExceptionPreformatted(
       case NOT_FOUND:
         if (message != null && message.contains("Session not found")) {
           return new SessionNotFoundException(token, message, cause);
+        } else if (message != null && DATABASE_NOT_FOUND_MSG_PATTERN.matcher(message).matches()) {
```

Comment:
Why use a regex matching instead of `message.contains()`? I mean that it is very sensitive to any small text/format change in the message.

Reply:
The reason I added the extensive regex matching for this exception is that it could be thrown by basically any RPC, and I worry about situations where the text 'Database not found' is used as part of the input, and then also returned by the server as part of an error message. I also added an integration test to ensure that the matching does stay in sync with the backend. But maybe I'm a little bit too paranoid here...